[2025-08-14T20:08:56.874+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [queued]>
[2025-08-14T20:08:56.881+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [queued]>
[2025-08-14T20:08:56.881+0000] {taskinstance.py:2170} INFO - Starting attempt 7 of 8
[2025-08-14T20:08:56.892+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): fetch_historical_data> on 2025-08-14 03:48:52+00:00
[2025-08-14T20:08:56.895+0000] {standard_task_runner.py:60} INFO - Started process 25665 to run task
[2025-08-14T20:08:56.898+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'stock_market_batch_pipeline', 'fetch_historical_data', 'manual__2025-08-14T03:48:52+00:00', '--job-id', '22', '--raw', '--subdir', 'DAGS_FOLDER/stock_market_batch_dag.py', '--cfg-path', '/tmp/tmp3gz2qak5']
[2025-08-14T20:08:56.900+0000] {standard_task_runner.py:88} INFO - Job 22: Subtask fetch_historical_data
[2025-08-14T20:08:56.952+0000] {task_command.py:423} INFO - Running <TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [running]> on host 0181ae2045ff
[2025-08-14T20:08:57.031+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Basanth Kumar' AIRFLOW_CTX_DAG_ID='stock_market_batch_pipeline' AIRFLOW_CTX_TASK_ID='fetch_historical_data' AIRFLOW_CTX_EXECUTION_DATE='2025-08-14T03:48:52+00:00' AIRFLOW_CTX_TRY_NUMBER='7' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-14T03:48:52+00:00'
[2025-08-14T20:08:57.035+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-08-14T20:08:57.037+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n    unset PYTHONNOUSERSITE\n    python /opt/***/dags/scripts/batch_data_producer.py 2025-08-14\n    ']
[2025-08-14T20:08:57.047+0000] {subprocess.py:86} INFO - Output:
[2025-08-14T20:08:58.151+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-08-14T20:08:58.158+0000] {subprocess.py:93} INFO -   File "/opt/***/dags/scripts/batch_data_producer.py", line 8, in <module>
[2025-08-14T20:08:58.159+0000] {subprocess.py:93} INFO -     import yfinance as yf
[2025-08-14T20:08:58.160+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'yfinance'
[2025-08-14T20:08:58.236+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-08-14T20:08:58.267+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-08-14T20:08:58.275+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=stock_market_batch_pipeline, task_id=fetch_historical_data, execution_date=20250814T034852, start_date=20250814T200856, end_date=20250814T200858
[2025-08-14T20:08:58.294+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 22 for task fetch_historical_data (Bash command failed. The command returned a non-zero exit code 1.; 25665)
[2025-08-14T20:08:58.310+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-14T20:08:58.353+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
